### DISCUSSION TITLE
Labeling tests

### DISCUSSION THREAD

**Discussion by chenrui333** [Orange-OpenSource/hurl#1403] [Ideas]
👋 I wonder if it is a good idea to introduce some attribute/label for test grouping, like for example, some tests cannot be run on CI due to Cloudflare integrity check failure, and some tests should be run at different frequencies. Let me know if that makes sense. Thanks!

**Comment by SilenLoc** [comment]
I use just command runner recipes and folders to group and slice the tests depending on env vars. But a baked-in solution would be cool.

**Reply by chenrui333** [reply]
yeah, the folder thing is what i am using right now as well, but the builtin labeling would be helpful with running and reporting. :)

**Comment by serle** [comment]
Hurl seems like a great tool to easily setup integration tests, however it is missing test names, test config (easily turn on and off tests with given labels), test hierarchies (maybe through folder structure) and a test runner cli tool which could inject variables, manage test concurrency and print out a compact test reports that can be consumed by ci.

**Reply by kiskit** [reply]
Test naming seems to be still compatible with the minimalist approach that hurl seems to favor and would be great. Something that gets repeated when running requests so we can know which one fails.

**Comment by fabricereix** [comment]
I have one remark relating to test semantics.
Tests in Hurl can be thought at different levels:
1) Running several Hurl files, each Hurl file can be considered as a test.
2) Within a Hurl file, each HTTP request can be considered as a test.
3) Within an Assert Section, each assert can be considered as a test.

We can add naming/metadata for these 3 test levels.
It's probably better to be more explicit when discussing about tests in Hurl.
We could stick to Hurl file, Request and Assert rather than tests.
